cloud/aws-common/src/main/java/org/apache/druid/common/aws/AWSClientUtil.java

```diff
@@ -54,6 +89,19 @@ public static boolean isClientExceptionRecoverable(AmazonClientException excepti
       return true;
     }
 
-    return RetryUtils.isClockSkewError(exception);
+    if (RetryUtils.isClockSkewError(exception)) {
+      return true;
+    }
+
+    if (exception instanceof MultiObjectDeleteException) {
+      MultiObjectDeleteException multiObjectDeleteException = (MultiObjectDeleteException) exception;
+      for (MultiObjectDeleteException.DeleteError error : multiObjectDeleteException.getErrors()) {
+        if (RECOVERABLE_ERROR_CODES.contains(error.getCode())) {
```

Comment:
This may be the status code and not errorCode, if so weird that it is a string and not ant integer. Confirming

Reply:
nvm, confirmed this is a string